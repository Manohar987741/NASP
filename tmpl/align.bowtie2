#!/bin/bash
# Read files may be .gz or .bz2
##### ERROR MESSAGE: SAM/BAM/CRAM file ${bam} is malformed: SAM file doesn't have any read groups defined in the header.
# The GATK no longer supports SAM files without read groups
# SOLUTION: Add --rg and --rg-id flags

# shellcheck source=./common.vars
source ${workdir}/job_scripts/common.vars


set_read1_and_read2_and_sample_name_and_bam "bowtie2"
mkdir -p "${workdir}/bams"
tmpdir=$(mktemp -d --tmpdir="${workdir}/bams")
trap "{
    rm -rf $tmpdir &
    eval ${snpcaller_release_cmd:?} &
    wait
}" EXIT


if [ -n "$read2" ]; then
	single_or_paired_read_arg="-1 $read1 -2 $read2"
else
	single_or_paired_read_arg="-U $read1"
fi


samtools=$(get_program_path "samtools")
if "$samtools" --version; then
    # Post samtools v1.0 syntax
    samtools_sort="$samtools sort -T $tmpdir -@ $ncpu -o $bam"
    printf "Using samtools post v1.0 syntax: %s" $samtools_sort
else
    # Pre samtools v1.0 syntax
    samtools_sort="$samtools sort -@ $ncpu - ${bam%.*}"
    printf "Using samtools pre v1.0 syntax: %s" $samtools_sort
fi

$(get_program_path "bowtie2") \
	--rg-id "${sample_name}" \
	--rg "SM:${sample_name}" \
	-x "${reference%.*}" \
	${single_or_paired_read_arg} \
| "$samtools" view -S -b -h - \
| $samtools_sort


"$samtools" index "$bam"
