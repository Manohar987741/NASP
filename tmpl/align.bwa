#!/bin/bash

# shellcheck source=./common.vars
source ${workdir}/job_scripts/common.vars


set_read1_and_read2_and_sample_name_and_bam "bwa"
mkdir -p "${workdir}/bams"
tmpdir=$(mktemp -d --tmpdir=${workdir}/bams)
trap '{
    rm -rf $tmpdir &
    eval "${snpcaller_release_cmd:?}" &
    wait
}' EXIT

samtools=$(get_program_path "samtools")
if "$samtools" --version; then
    # Post samtools v1.0 syntax
    samtools_sort="$samtools sort -T $tmpdir -@ $ncpu -o $bam"
    printf "Using samtools post v1.0 syntax: %s" $samtools_sort
else
    # Pre samtools v1.0 syntax
    samtools_sort="$samtools sort -@ $ncpu - ${bam%.*}"
    printf "Using samtools pre v1.0 syntax: %s" $samtools_sort
fi


bwa=$(get_program_path "bwa")
if [ -n "$read2" ]; then
	$bwa mem \
		-R "@RG\\tID:${sample_name}\\tSM:${sample_name}" \
		-t "$ncpu" \
		"$reference" \
		"$read1" \
		"$read2" \
    | "$samtools" view -S -b -h - \
	| $samtools_sort
else
	$bwa samse -r "@RG\\tID:${sample_name}\\tSM:name" "$reference" \
		<(bwa aln "$reference" "$read1" -t "$ncpu" -f "${workdir}/bams/${sample_name}.sai") \
		"$read1" \
    | "$samtools" view -S -b -h - \
	| $samtools_sort
fi

$(get_program_path "samtools") index "$bam"
