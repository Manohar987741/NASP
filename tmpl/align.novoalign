#!/bin/bash

# shellcheck source=./common.vars
source ${workdir}/job_scripts/common.vars


set_read1_and_read2_and_sample_name_and_bam "bwa"
tmpdir=$(mktemp -d --tmpdir=${workdir})
trap '{
    rm -rf $tmpdir &
    eval "${snpcaller_release_cmd:?}" &
    wait
}' EXIT

samtools=$(get_program_path "samtools")
if "$samtools" --version; then
    # Post samtools v1.0 syntax
    samtools_sort="$samtools sort -T $tmpdir -@ $ncpu -o $bam"
    printf "Using samtools post v1.0 syntax: %s" $samtools_sort
else
    # Pre samtools v1.0 syntax
    samtools_sort="$samtools sort -@ $ncpu - ${bam%.*}"
    printf "Using samtools pre v1.0 syntax: %s" $samtools_sort
fi


$(get_program_path "novoalign") \
	-r all \
	-f $read1 $read2 \
	-c $ncpu \
	-o SAM "@RG\\tID:${sample_name}\\tSM:${sample_name}" \
	-d ${reference}.idx \
| "$samtools" view -S -b -h - \
| $samtools_sort


$(get_program_path "samtools") index "$bam"
