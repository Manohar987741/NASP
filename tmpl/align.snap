#!/bin/bash
#
# Add -M option by default to make SNAP generate traditional CIGAR strings in the BAM output to be more compatible with SNP callers
#
# The two FASTQ files are not the same size! Make sure they contain
# the same reads in the same order, without comments.
# DEVTEAM: Allowing run, but you need to run on only one thread because the file splitter
# Doesn't understand how to deal with files that don't match byte-for-byte.

# TODO: check fastq file size and disable multithreading if they don't match
# shellcheck source=./common.vars
source ${workdir}/job_scripts/common.vars


set_read1_and_read2_and_sample_name_and_bam "bwa"
tmpdir=$(mktemp -d --tmpdir=${workdir})
trap '{
    rm -rf $tmpdir &
    eval "${snpcaller_release_cmd:?}" &
    wait
}' EXIT INT TERM HUP


fifo=${tmpdir}/fifo
mkfifo $fifo

read1_basename=${read1##*/}
read2_basename=${read2##*/}

# Path to the read files if they needed to be decompressed before aligning
read1_uncompressed=${tmpdir}/${read1_basename%.*}
read2_uncompressed=${tmpdir}/${read2_basename%.*}


samtools=$(get_program_path "samtools")
if "$samtools" --version; then
    # Post samtools v1.0 syntax
    samtools_sort="$samtools sort -T $tmpdir -@ $ncpu -o $bam"
    printf "Using samtools post v1.0 syntax: %s" $samtools_sort
else
    # Pre samtools v1.0 syntax
    samtools_sort="$samtools sort -@ $ncpu - ${bam%.*}"
    printf "Using samtools pre v1.0 syntax: %s" $samtools_sort
fi


snap=$(get_program_path "snap")
if [ -n "$read2" ]; then
	if [[ "$read1" = *.gz ]]; then
		gunzip -c $read1 > ${read1_uncompressed} &
		gunzip -c $read2 > ${read2_uncompressed} &
		read1=${read1_uncompressed}
		read2=${read2_uncompressed}
		wait # wait for the gunzip jobs to finish
	elif [[ "$read1" = *.bz2 ]]; then
		bunzip2 -c $read1 > ${read1_uncompressed} &
		bunzip2 -c $read2 > ${read2_uncompressed} &
		read1=${read1_uncompressed}
		read2=${read2_uncompressed}
		wait # wait for the bunzip2 jobs to finish
	fi

    "$samtools" view -S -b -h "$fifo" \
    $samtools_sort &
	$snap paired ${reference%.*} ${read1} ${read2} -o ${fifo} -M
else
	if [[ "$read1" = *.gz ]]; then
		gunzip -c $read1 > ${read1_uncompressed}
		read1=${read1_uncompressed}
	elif [[ "$read" = *.bz2 ]]; then
		bunzip2 -c $read1 > ${read1_uncompressed}
		read1=${read1_uncompressed}
	fi

    "$samtools" view -S -b -h "$fifo" \
    $samtools_sort &
	$snap single ${reference%.*} ${read1} -o ${fifo} -M
fi

wait # wait for the samtools view/sort job to finish

$(get_program_path "samtools") index ${bam}
